### DISCUSSION TITLE
Budibase: how to fix some errors in a form

### DISCUSSION THREAD

**Discussion by dianaspetcu** [Budibase/budibase#10493] [Help]
Hi team,
I have 3 questions from a form review:
1. In the field "name"  I have add a custom validation rule that you can only use alphabetic characters. Well, when I test it that field is showing me the error "you can use only alphabetic" even if I write the correct name or a name with number. (The error is not disappearing) How can I make the error disappear from the form when I put the correct name?
2. How can you setup the field name "phone" to start the phone number with 0?
3. What "define action" should I put for the button to clear the form after I have completed it? I have selected the "save", "validate" and "clear".

Another question is from automation part:
1. Why do I have an error about a field (max length 6 numbers) if I put the number 2000? 
{
  "success": false,
  "response": "{\"validation\":{\"Km\":[\"must be less than or equal to 6\"]}}"
}

**Comment by ConorWebb96** [comment]
Hey @dianaspetcu,

1. Could you provide a screenshot of the field and its custom validation so I can re-create? From reading your message I can't really tell what the issue is.
2. There currently is bug around this: https://github.com/Budibase/budibase/issues/10118, normally all you would need to do is add 0 to the default value and it will appear before it. However the ui currently hides zero if its the only value.
3. All you need is the `Clear Form` action and selecting the correct form and this should work out of the box for you.

For your last question have you set a max value within the data area for that column? That could be whats through this error. Possibly check any validation on the field itself as well.

I hope this helps!

**Reply by dianaspetcu** [reply]
![error on name field](https://user-images.githubusercontent.com/132670969/237062428-13688326-edc5-4fe5-b0c0-e73a317f85d9.jpg)
![custom validation rule for name](https://user-images.githubusercontent.com/132670969/237062462-454e978a-1e42-4ed1-9b5b-238e55d3746e.jpg)

Hello @ConorWebb96,
Thanks for the answers :)

When I am writing the correct name is ok. If I delete the name that I entered and put a number the error is appearing. After that if I correct with another name, the error is remaining there. Why the error is not disappearing after I corrected the field?

For the last question: If I do not setup a max value of the field, all is good. If I setup a max value (6 numbers) and I entered 4 numbers than the error is appearing.
And for all form as well. If the form does not have setup for some fields custom validation rule, the button with "validate, save and clear form" is working. If, for example, the error is persisting in the field name, the form cannot be saved and cleared.

**Reply by melohagan** [reply]
Hey @dianaspetcu I think your regex looks wrong. Should it not be: `^[a-zA-Z\s]*$`
